File: pypy/module/array/benchmark/intimgtst.c
void trans(unsigned char *img, unsigned int *intimg);

void main2() {
  unsigned char img[640*480];
  unsigned int intimg[640*480];

  for (int l=0; l<500; l++) trans(img,intimg);
}

void transf(double *img, double *intimg);
void main() {
  double img[640*480];
  double intimg[640*480];

  for (int l=0; l<500; l++) transf(img,intimg);
}

File: pypy/module/array/benchmark/intimgtst.py
#!/usr/bin/python
from array import array

def f(img, intimg):
    l=0
    i=640    
    while i<640*480:
        l+=img[i]
        intimg[i]=intimg[i-640]+l
        i+=1
    return l



img=array('d','\x00'*640*480*8)
intimg=array('d','\x00'*640*480*8)

for l in range(500): f(img, intimg)
